mavros_extras/src/plugins/obstacle_distance_3d.cpp

```diff
@@ -0,0 +1,113 @@
+/**
+ * @brief Obstacle Distance 3D plugin
+ * @file obstacle_3d.cpp
+ * @author Zeke Sarosi
+ *
+ * @addtogroup plugin
+ * @{
+ */
+/*
+ * Copyright 2025 Zeke Sarosi.
+ *
+ * This file is part of the MAVROS package and subject to the license terms
+ * in the top-level LICENSE file of the MAVROS repository.
+ */
+
+ #include <cmath> // For std::isnan
+
+ #include "mavros/mavros_uas.hpp"
+ #include "mavros/plugin.hpp"
+ #include "mavros/plugin_filter.hpp"
+ 
+  #include "mavros_msgs/msg/obstacle3_d.hpp"
+ 
+ namespace mavros
+ {
+ namespace extra_plugins
+ {
+ using namespace std::placeholders;
+ 
+ /**
+  * @brief Plugin to handle sending OBSTACLE_DISTANCE_3D MAVLink messages.
+  * @plugin obstacle_distance_3d
+  *
+   * Subscribes to a mavros_msgs/Obstacle3D message and sends the
+  * data to the flight controller to report the position of a single obstacle.
+  */
+ class Obstacle3DPlugin : public plugin::Plugin
+ {
+ public:
+     explicit Obstacle3DPlugin(plugin::UASPtr uas_)
+     : Plugin(uas_, "obstacle_distance_3d")
```

Comment:
Better to keep consistent class name and node name.
So i think better to name a class `ObstacleDistance3DPlugin`.